modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rule-builder/editor/Actions.es.js

```diff
@@ -41,11 +41,19 @@ const ActionContentCalculate = ({
 		link: location.origin + functionsURL,
 	});
 
-	const options = useMemo(() => {
-		return fields.filter(
-			({fieldName, type}) => type === 'numeric' && fieldName !== target
-		);
-	}, [fields, target]);
+	const options = useMemo(
+		() =>
+			fields.filter(
+				({fieldName, type}) =>
+					type === 'numeric' && fieldName !== target
+			),
+		[fields, target]
+	);
+
+	const numericFields = useMemo(
+		() => fields.filter(({type}) => type === 'numeric'),
+		[fields]
+	);
```

Comment:
Why not 
```
const numericFields = useMemo(
		() => fields.filter(({type}) => type === 'numeric'),
		[fields]
	);
	
	const options = useMemo(
		() =>
			options.filter(
				({fieldName}) =>
					fieldName !== target
			),
		[options]
	);

```

because We are filtering just numeric fields and working with just numeric fields will contain less elements to iterate than iterating over fields again

Reply:
Good catch, makes sense.